### PR TITLE
stats: Add PR stats plugin

### PR DESCRIPTION
**Pull request by meetmangukiya**
Closes https://github.com/coala/corobo/issues/224

# Reviewers Checklist

- [ ] Appropriate logging is done.
- [ ] Appropriate error responses.
- [ ] Handle every possible exception.
- [ ] Make sure there is a docstring in the command functions. Hint: Lookout for
  `botcmd` and `re_botcmd` decorators.
- [ ] See that 100% coverage is there.
- [ ] See to it that mocking is not done where it is not necessary.

### CODE REVIEW

**Comment by meetmangukiya** [comment]
# BLOCKER - https://github.com/errbotio/errbot/issues/1086

File: plugins/labhub.py [hunk 1]
```diff
@@ -326,3 +328,50 @@ def eligible(user, iss):
                 yield ('The issue is already assigned to someone. Please '
                        'check if the assignee is still working on the issue, '
                        'if not, you should ask for reassignment.')
+
+    @staticmethod
+    def community_state(pr_count: dict):
```
Comment (Udayan12167) [inline plugins/labhub.py]: Is this overall or just for some timeperiod?

Comment (Udayan12167) [reply plugins/labhub.py]: Oh I see in the past x hours is it? Depending on the input?

File: plugins/labhub.py [hunk 2]
```diff
@@ -326,3 +328,50 @@ def eligible(user, iss):
...
+        pr_count = dict()
+        start = time.time()
+        for count, (name, repo) in enumerate(self.gh_repos.items(), 1):
+            pr_count[name] = len(list(repo.search_mrs(
+                                         created_after=(datetime.datetime.now()
+                                                        - datetime.timedelta(
+                                                              hours=int(hours)))
+                             )))
+            # handle github rate limiting
+            if (count % 30 == 0):
```
Comment (Udayan12167) [inline plugins/labhub.py]: Why %30?

**Comment by Udayan12167** [comment]
ack https://github.com/coala/corobo/pull/277/commits/8020287179bd9ea2db848c0b4265b3bdb16d0f3e

**Comment by Udayan12167** [comment]
ack https://github.com/coala/corobo/pull/277/commits/e61571d86d9d9b03ca589d2867ad9c9687bb80e9

**Comment by Udayan12167** [comment]
ack https://github.com/coala/corobo/pull/277/commits/e61571d86d9d9b03ca589d2867ad9c9687bb80e9

**Comment by Udayan12167** [comment]
ack https://github.com/coala/corobo/pull/277/commits/6a2180918f1ce13e1b50ba952e9f6c80c4373da9

**Comment by Udayan12167** [comment]
ack https://github.com/coala/corobo/pull/277/commits/98ebcbfc891a842f871ef9d3f45ac39a505899ed

File: plugins/labhub.py (hunk 2 above)
Comment (meetmangukiya) [reply plugins/labhub.py]: Because github allows 30 requests per minute for search endpoints

File: plugins/labhub.py (hunk 1 above)
Comment (meetmangukiya) [reply plugins/labhub.py]: Yes, depending on the hours given in the command. `corobo pr stats 10hrs`, we consider last 10 hours

Comment (meetmangukiya) [reply plugins/labhub.py]: Yes, depending on the hours given in the command. `corobo pr stats 10hrs`, we consider last 10 hours

**Comment by meetmangukiya** [comment]
@rultor merge

**Pull request merged**